File: truevfs-ext/truevfs-ext-logging/src/main/scala/net/java/truevfs/ext/logging/LogResource.scala
/*
 * Copyright (C) 2005-2015 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.logging

import org.slf4j._

private trait LogResource {

  def logger: Logger

  def log(message: String, parameter: AnyRef): Unit = {
    logger debug (message, parameter)
    if (logger.isTraceEnabled) {
      logger.trace("Stack trace:", new Throwable)
    }
  }
}

File: truevfs-ext/truevfs-ext-logging/src/main/java/net/java/truevfs/ext/logging/LogResource.java
/*
 * Copyright © 2005 - 2021 Schlichtherle IT Services.
 * All rights reserved. Use is subject to license terms.
 */
package net.java.truevfs.ext.logging;

import lombok.val;
import org.slf4j.Logger;

interface LogResource {

    Logger logger();

    default void log(final String message, final Object parameter) {
        val logger = logger();
        logger.debug(message, parameter);
        if (logger.isTraceEnabled()) {
            logger.trace("Stack trace:", new Throwable());
        }
    }
}
